File: poker-after-dark-domain/src/main/java/com/slusarz/pokerafterdark/domain/tournament/TournamentParticipant.java
package com.slusarz.pokerafterdark.domain.tournament;

import com.slusarz.pokerafterdark.domain.earnings.Earnings;
import com.slusarz.pokerafterdark.domain.player.PlayerId;
import com.slusarz.pokerafterdark.domain.validation.ValidationError;
import lombok.Value;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

@Value(staticConstructor = "of")
public class TournamentParticipant {

    @Valid
    @NotNull(message = ValidationError.MANDATORY_PLAYER_ID)
    private PlayerId playerId;

    @Valid
    @NotNull(message = ValidationError.MANDATORY_EARNINGS)
    private Earnings earnings;

    @Valid
    @NotNull(message = ValidationError.MANDATORY_PLACE)
    private Place place;

}

File: poker-after-dark-web/src/app/shared/models/tournament/tournament-participant.ts
import { PlayerInTournament } from "./player-in-tournament";

export class TournamentParticipant {

    playerId: string;

    earnings: number;

    place: number;

    constructor(player: PlayerInTournament) {
        this.playerId = player.player.id;
        this.earnings = player.earnings;
        this.place = player.place;
    }

}
